src/main/java/frc/team2412/robot/commands/index/IndexCommand.java

```diff
@@ -4,27 +4,13 @@
 import frc.team2412.robot.subsystem.IndexSubsystem;
 
 public class IndexCommand extends SequentialCommandGroup {
-    private final IndexSubsystem subsystem;
 
     public IndexCommand(IndexSubsystem subsystem) {
-        this.subsystem = subsystem;
-        addRequirements(subsystem);
-    }
-
-    @Override
-    public void initialize() {
-        // move the ball into the first sensor
-        addCommands(new IndexMoveBallToFirstSensor(subsystem));
-
-        // if both spaces are open
         if (!subsystem.feederSensorHasBallIn()) {
```

Comment:
Note that this check will run when the command is first created, and not when the command first runs (because the check was moved from `initialize()` to the constructor). So if the command is created when the robot turns on, it'll check if there's a ball then instead of when the command starts executing (like when drive team presses a button).

One solution would be to use [`ConditionalCommand`](https://docs.wpilib.org/en/stable/docs/software/commandbased/convenience-features.html#conditionalcommand) instead and choose between two command groups, as this will wait until the command starts running before checking `feederSensorHasBallIn()`.